synapse/storage/databases/main/events_worker.py

```diff
@@ -1979,3 +1979,27 @@ async def is_partial_state_event(self, event_id: str) -> bool:
             desc="is_partial_state_event",
         )
         return result is not None
+
+    async def get_partial_state_events_batch(self, room_id: str) -> List[str]:
+        """Get a list of events in the given room that have partial state"""
+        return await self.db_pool.runInteraction(
+            "get_partial_state_events_batch",
+            self._get_partial_state_events_batch_txn,
+            room_id,
+        )
+
+    @staticmethod
+    def _get_partial_state_events_batch_txn(
+        txn: LoggingTransaction, room_id: str
+    ) -> List[str]:
+        txn.execute(
+            """\
```

Comment:
I don't think we need the backslash (and don't usually put one)?

```suggestion
            """
```